crypto/rsa/rsa_local.h

```diff
@@ -10,6 +10,7 @@
 #ifndef OSSL_CRYPTO_RSA_LOCAL_H
 #define OSSL_CRYPTO_RSA_LOCAL_H
 
+#include "crypto/rsa.h"
```

Comment:
This look like a errant change?

Reply:
Unfortunately it is necessary to work around a bug in the standard library. See the commit message of 6faf44f53f322ffccd0359d68bb4f345fdbaad69.

Reply:
The commit message contains a link to the publically unavailable area.

Reply:
@beldmit Anyone can create an account on their bugtracker; it's the same account credentials registered when [downloading C++Builder Community Edition](https://www.embarcadero.com/products/cbuilder/starter/free-download). I didn't have to purchase anything to do so.

Reply:
Please add a comment here explaining the situation.  People don't always have direct access to git history.

Reply:
I tried reproducing the problem in my branch without this extra `#include`, and it reproduced:

```
crypto\rsa\rsa_meth.c:
In file included from crypto\rsa\rsa_meth.c:17:
In file included from crypto\rsa/rsa_local.h:13:
In file included from include\internal/refcount.h:22:
c:\program files (x86)\embarcadero\studio\20.0\include\dinkumware64\stdatomic.h:77:23: error:
      unknown type name 'char16_t'
_ATOMIC_DEF(char16_t, char16_t)
                      ^
c:\program files (x86)\embarcadero\studio\20.0\include\dinkumware64\stdatomic.h:78:23: error:
      unknown type name 'char32_t'
_ATOMIC_DEF(char32_t, char32_t)
                      ^
In file included from crypto\rsa\rsa_meth.c:17:
In file included from crypto\rsa/rsa_local.h:13:
include\internal/refcount.h:35:12: error: member reference base type
      '_Atomic(int)' is not a structure or union
    *ret = atomic_fetch_add_explicit(val, 1, memory_order_relaxed) + 1;
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
c:\program files (x86)\embarcadero\studio\20.0\include\dinkumware64\stdatomic.h:244:47: note:
      expanded from macro 'atomic_fetch_add_explicit'
        _GEN_SUF(_Myatomic_fetch_add_explicit, (pobj)->_Atom)(pobj, val, ord)
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~
c:\program files (x86)\embarcadero\studio\20.0\include\dinkumware64\stdatomic.h:117:12: note:
      expanded from macro '_GEN_SUF'
        _Generic((VAL) + (uint_least8_t)0, \
                  ^~~
In file included from crypto\rsa\rsa_meth.c:17:
In file included from crypto\rsa/rsa_local.h:13:
include\internal/refcount.h:51:12: error: member reference base type
      '_Atomic(int)' is not a structure or union
    *ret = atomic_fetch_sub_explicit(val, 1, memory_order_relaxed) - 1;
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
c:\program files (x86)\embarcadero\studio\20.0\include\dinkumware64\stdatomic.h:249:47: note:
      expanded from macro 'atomic_fetch_sub_explicit'
        _GEN_SUF(_Myatomic_fetch_sub_explicit, (pobj)->_Atom)(pobj, val, ord)
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~
c:\program files (x86)\embarcadero\studio\20.0\include\dinkumware64\stdatomic.h:117:12: note:
      expanded from macro '_GEN_SUF'
        _Generic((VAL) + (uint_least8_t)0, \
                  ^~~
4 errors generated.
```

However, when I tried with merging the current "master" branch (52c6c12c1) into a detached checkout of my branch, the problem went away. My guess is that the problem resolved itself due to [this change made to `include/internal/refcount.h`](https://github.com/openssl/openssl/commit/a7e6a3d8ef4eea0e211441392b80a7acfd4a16b2#diff-1acae6682cfc2672070ea1e2232573e9c60acf42c58d98413ee7fc5ddf02068bR12) made in "master", which [causes `<inttypes.h>` to be included via `openssl/e_os2.h`](https://github.com/openssl/openssl/blob/master/include/openssl/e_os2.h#L234) before we get to including `<stdatomic.h>` later on in `include/internal/refcount.h` (line 23/24).

So I'm undoing this change, but keeping the bug report to Embarcadero.

Reply:
Sounds like a good enough explanation.  The bug report to Embarcadero sounds reasonable...  basically, it should be possible to include `<stdatomic.h>` alone, I assume that's about the gist of it...

Reply:
Yes, the bug report is that a one-line C file with `#include <stdatomic.h>` doesn't compile with their Clang-based compiler. Any standard C header should be compile-able by itself.